device/iot-device-samples/device-connection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceConnectionSample.java

```diff
@@ -0,0 +1,119 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.ClientOptions;
+import com.microsoft.azure.sdk.iot.device.DeviceClient;
+import com.microsoft.azure.sdk.iot.device.IotHubClientProtocol;
+import com.microsoft.azure.sdk.iot.service.auth.IotHubConnectionStringBuilder;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+import com.microsoft.azure.sdk.iot.service.methods.DirectMethodsClient;
+import com.microsoft.azure.sdk.iot.service.registry.Device;
+import com.microsoft.azure.sdk.iot.service.registry.RegistryClient;
+import com.microsoft.azure.sdk.iot.service.twin.Twin;
+import com.microsoft.azure.sdk.iot.service.twin.TwinClient;
+
+import java.io.IOException;
+import java.util.UUID;
+
+public class DeviceConnectionSample
+{
+    private static final String desiredPropertyKey = "someDesiredPropertyKey";
+
+    // This sample is only useful for MQTT, MQTT_WS, AMQPS, and AMQPS_WS,  since they are the only stateful connection protocols.
+    public static void main(String[] args) throws InterruptedException, IOException, IotHubException
+    {
+        String iotHubConnectionString = System.getenv("IOTHUB_CONNECTION_STRING");
+        RegistryClient registryClient = new RegistryClient(iotHubConnectionString);
+        TwinClient twinClient = new TwinClient(iotHubConnectionString);
+        DirectMethodsClient directMethodsClient = new DirectMethodsClient(iotHubConnectionString);
+        Device d = registryClient.addDevice(new Device("timtay-longhaul-test-" + UUID.randomUUID()));
+        final String deviceConnectionString = getDeviceConnectionString(iotHubConnectionString, d);
+        final IotHubClientProtocol protocol = IotHubClientProtocol.AMQPS;
+        final ClientOptions clientOptions = ClientOptions.builder().keepAliveInterval(10).build();
+        final DeviceClient deviceClient = new DeviceClient(deviceConnectionString, protocol, clientOptions);
+        DeviceClientManager deviceClientManager = new DeviceClientManager(deviceClient);
+
+        Thread deviceThread = new Thread(() -> deviceClientManager.run());
+        deviceThread.start();
+
+        Twin twin = twinClient.get(d.getDeviceId());
+        System.out.println("Received initial twin from twin service client");
+        System.out.println(twin);
+
+        while (deviceThread.isAlive())
+        {
+            Thread.sleep(10000);
+
+            try
+            {
+                directMethodsClient.invoke(d.getDeviceId(), "someMethodName");
```

Comment:
Is this sample making use of the service SDKs too? I think some code comments would be helpful in this file.